java/src/com/ibm/streamsx/rest/StreamingAnalyticsServiceV1.java

```diff
@@ -0,0 +1,209 @@
+import org.apache.http.entity.mime.content.StringBody;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.ibm.streamsx.topology.context.remote.RemoteContext;
+import com.ibm.streamsx.topology.internal.context.remote.DeployKeys;
+
+/**
+ * Implementation of StreamingAnalyticsService for Version 1.
+ * <p>
+ * This is cut & paste of the code from the original uses. 
+ */
+class StreamingAnalyticsServiceV1 extends AbstractStreamingAnalyticsService {
+    StreamingAnalyticsServiceV1(JsonObject service) {
+        super(service);
+        // Authorization header never changes in V1 once set
+        setAuthorization(StreamsRestUtils.createBasicAuth(credentials));
+    }
+
+    @Override
+    protected String getStatusUrl(CloseableHttpClient httpClient) {
+        StringBuilder sb = new StringBuilder(500);
+        sb.append(jstring(credentials, "rest_url"));
+        sb.append(jstring(credentials, "status_path"));
+        return sb.toString();
+    }
+
+    @Override
+    protected String getJobSubmitUrl(CloseableHttpClient httpClient, File bundle)
+            throws UnsupportedEncodingException {
+        StringBuilder sb = new StringBuilder(500);
+        sb.append(jstring(credentials, "rest_url"));
+        sb.append(jstring(credentials, "jobs_path"));
+        sb.append("?");
+        sb.append("bundle_id=");
+        sb.append(URLEncoder.encode(bundle.getName(), StandardCharsets.UTF_8.name()));
+        return sb.toString();
+    }
+
+    @Override
+    protected String getJobSubmitUrl(JsonObject artifact)
+            throws UnsupportedEncodingException {
+        String artifactId = jstring(artifact, "id");
+        StringBuilder sb = new StringBuilder(500);
+        sb.append(jstring(credentials, "rest_url"));
+        sb.append(jstring(credentials, "jobs_path").replace("jobs", "builds"));
+        sb.append("?");
+        sb.append("artifact_id=");
+        sb.append(URLEncoder.encode(artifactId, StandardCharsets.UTF_8.name()));
+        return sb.toString();
+}
+
+    @Override
+    protected String getBuildsUrl(CloseableHttpClient httpClient){
+        String buildURL = jstring(credentials, "jobs_path").replace("jobs", "builds");
+        return jstring(credentials, "rest_url") + buildURL;
+    }
+
+    @Override
+    protected String getAuthorization() {
+        return authorization;
+    }
+
+    @Override
+    protected String getJobSubmitId() {
+        return "jobId";
+    }
+
+    @Override
+    protected JsonObject getBuild(String buildId, CloseableHttpClient httpclient,
+            String authorization) throws IOException {
+        String buildURL = getBuildsUrl(httpclient) + "?build_id="
+            + URLEncoder.encode(buildId, StandardCharsets.UTF_8.name());
+        HttpGet httpget = new HttpGet(buildURL);
+        httpget.addHeader("accept", ContentType.APPLICATION_JSON.getMimeType());
```

Comment:
This is redundant. You already specify the `APPLICATION_JSON` in the getGsonResponse method.